metrics/metrics.go

```diff
@@ -0,0 +1,93 @@
+package metrics
+
+import (
+	"context"
+	"net"
+	"net/http"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+	"go.opentelemetry.io/otel/sdk/metric"
+)
+
+var (
+	log = logging.Logger("metrics")
+)
+
+type Metrics struct {
+	Exporter *prometheus.Exporter
+
+	GetValueKeysLatency      syncint64.Histogram
+	GetMetadataLatency       syncint64.Histogram
+	PutMultihashBatchLatency syncint64.Histogram
+	PutMetadataLatency       syncint64.Histogram
+
+	s *http.Server
+}
+
+func New(metricsAddr string) (*Metrics, error) {
+
+	m := &Metrics{}
```

Comment:
Style comment: I find the code reads better if `m` is defined as a zero-value variable, and then its pointer is returned later in line 70, unless, you need to initialise fields in the `Metrics struct. 
```suggestion
	var m Metrics
```